rgw/v2/tests/s3cmd/test_large_object_gc.py

```diff
@@ -0,0 +1,173 @@
+    Delete bucket
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import sys
+import time
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib import resource_op
+from v2.lib.admin import UserMgmt
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.lib.s3cmd import auth as s3_auth
+from v2.tests.s3_swift import reusable
+from v2.tests.s3cmd import reusable as s3cmd_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+
+
+def test_exec(config):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    umgmt = UserMgmt()
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+    # preparing data
+    user_name = resource_op.create_users(no_of_users_to_create=1)[0]["user_id"]
+    tenant = "tenant"
+    tenant_user_info = umgmt.create_tenant_user(
+        tenant_name=tenant, user_id=user_name, displayname=user_name
+    )
+    umgmt.create_subuser(tenant_name=tenant, user_id=user_name)
+
+    ip_and_port = s3cmd_reusable.get_rgw_ip_and_port()
+    s3_auth.do_auth(tenant_user_info, ip_and_port)
+
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    # Create a bucket
+    s3cmd_reusable.create_bucket(bucket_name)
+    log.info(f"Bucket {bucket_name} created")
+
+    # Upload a 2GB file to bucket
+    uploaded_file_info = s3cmd_reusable.upload_file(bucket_name, file_size=2147483648)
+    uploaded_file = uploaded_file_info["name"]
+    uploaded_file_md5 = uploaded_file_info["md5"]
+    log.info(f"Uploaded file {uploaded_file} to bucket {bucket_name}")
+
+    if config.gc_verification is True:
+        log.info("making changes to ceph.conf")
+        config.rgw_gc_obj_min_wait = 5
+        ceph_conf.set_to_ceph_conf(
+            "global",
+            ConfigOpts.rgw_gc_obj_min_wait,
+            str(config.rgw_gc_obj_min_wait),
+        )
+        log.info("trying to restart services")
+        srv_restarted = rgw_service.restart()
+        time.sleep(30)
+        if srv_restarted is False:
+            raise TestExecError("RGW service restart failed")
+        else:
+            log.info("RGW service restarted")
+        log.info("download large object again to make gc list with shadow entries")
+
+        s3cmd_reusable.download_file(
+            bucket_name, uploaded_file, local_file_name="download1.img"
+        )
+        time.sleep(5)
+        downloaded_file1_md5 = utils.get_md5("download1.img")
```

Comment:
if we are not using the download file... it is good to remove it as the assert statement would leave residues and could potentially cause the system to run out of space. 

Have a cleanup to ensure the reliability of the system.

Reply:
Will add cleanup scripts. @rakeshgm , do we have method to cleanup files in TEST_DATA directory?

Reply:
@vivekanandan-k-rh  you can delete all of them. but check once if `io_info.yaml` is creating in that dir. my guess is that io_info.yaml wont be present. better to check. and add the cleanup code in utils.py so that we can call this function in other cases too.

Reply:
Done